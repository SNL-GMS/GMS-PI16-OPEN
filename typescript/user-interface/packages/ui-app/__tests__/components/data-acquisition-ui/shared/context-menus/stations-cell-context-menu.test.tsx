import { ContextMenu } from '@blueprintjs/core';
import React from 'react';

import {
  acknowledgeOnClick,
  DisabledStationSohContextMenu,
  getStationAcknowledgementMenuText,
  StationSohContextMenu,
  StationSohContextMenuItemProps,
  StationSohContextMenuProps
} from '../../../../../src/ts/components/data-acquisition-ui/shared/context-menus/stations-cell-context-menu';

// eslint-disable-next-line @typescript-eslint/no-var-requires, @typescript-eslint/no-require-imports
const Enzyme = require('enzyme');

describe('Station cell context menu', () => {
  const stationSohContextMenuProps: StationSohContextMenuProps = {
    stationNames: ['test', 'test2'],
    acknowledgeCallback: jest.fn()
  };

  const stationSohContextMenu = Enzyme.mount(
    // eslint-disable-next-line react/jsx-props-no-spreading
    <StationSohContextMenu {...stationSohContextMenuProps} />
  );
  const disabledStationSohContextMenu = Enzyme.mount(
    // eslint-disable-next-line react/jsx-props-no-spreading
    <DisabledStationSohContextMenu {...stationSohContextMenuProps} />
  );

  it('StationSohContextMenu can be created', () => {
    expect(stationSohContextMenu).toMatchSnapshot();
  });

  it('DisabledStationSohContextMenu can be created', () => {
    expect(disabledStationSohContextMenu).toMatchSnapshot();
  });
  it('getStationAcknowledgementMenuText to work as expected', () => {
    const stationNames = ['test', 'test1'];
    const withComment = false;
    const expectedResult = 'Acknowledge 2 stations';
    const result = getStationAcknowledgementMenuText(stationNames, withComment);
    expect(result).toEqual(expectedResult);
  });

  it('acknowledgeOnClick to work as expected', () => {
    const stationSohContextMenuItemProps: StationSohContextMenuItemProps = {
      ...stationSohContextMenuProps,
      disabled: false
    };
    const createSpy = jest
      .spyOn(ContextMenu, 'show')
      // eslint-disable-next-line no-console
      .mockImplementation(() => console.log('shown'));
    const mouseEvent: any = {
      preventDefault: jest.fn(),
      clientX: 5,
      clientY: 5
    };
    acknowledgeOnClick(mouseEvent, stationSohContextMenuItemProps);
    expect(createSpy).toHaveBeenCalled();
  });
});
