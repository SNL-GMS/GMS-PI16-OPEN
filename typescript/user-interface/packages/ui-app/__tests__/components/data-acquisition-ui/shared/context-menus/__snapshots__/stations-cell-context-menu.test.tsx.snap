// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Station cell context menu DisabledStationSohContextMenu can be created 1`] = `
<DisabledStationSohContextMenu
  acknowledgeCallback={[MockFunction]}
  stationNames={
    Array [
      "test",
      "test2",
    ]
  }
>
  <Blueprint3.Menu>
    <ul
      className="bp3-menu"
    >
      <AcknowledgeMenuItem
        acknowledgeCallback={[MockFunction]}
        disabled={true}
        stationNames={
          Array [
            "test",
            "test2",
          ]
        }
      >
        <Blueprint3.MenuItem
          className="acknowledge-soh-status"
          data-cy="acknowledge-without-comment"
          disabled={true}
          multiline={false}
          onClick={[Function]}
          popoverProps={Object {}}
          shouldDismissPopover={true}
          text="Acknowledge 2 stations"
        >
          <li
            className=""
          >
            <a
              className="bp3-menu-item bp3-disabled acknowledge-soh-status"
              data-cy="acknowledge-without-comment"
              tabIndex={-1}
            >
              <Blueprint3.Icon />
              <Blueprint3.Text
                className="bp3-fill"
                ellipsize={true}
              >
                <div
                  className="bp3-text-overflow-ellipsis bp3-fill"
                >
                  Acknowledge 2 stations
                </div>
              </Blueprint3.Text>
            </a>
          </li>
        </Blueprint3.MenuItem>
      </AcknowledgeMenuItem>
      <AcknowledgeWithCommentMenuItem
        acknowledgeCallback={[MockFunction]}
        disabled={true}
        stationNames={
          Array [
            "test",
            "test2",
          ]
        }
      >
        <Blueprint3.MenuItem
          className="acknowledge-soh-status"
          data-cy="acknowledge-with-comment"
          disabled={true}
          multiline={false}
          onClick={[Function]}
          popoverProps={Object {}}
          shouldDismissPopover={true}
          text="Acknowledge 2 stations with comment..."
        >
          <li
            className=""
          >
            <a
              className="bp3-menu-item bp3-disabled acknowledge-soh-status"
              data-cy="acknowledge-with-comment"
              tabIndex={-1}
            >
              <Blueprint3.Icon />
              <Blueprint3.Text
                className="bp3-fill"
                ellipsize={true}
              >
                <div
                  className="bp3-text-overflow-ellipsis bp3-fill"
                >
                  Acknowledge 2 stations with comment...
                </div>
              </Blueprint3.Text>
            </a>
          </li>
        </Blueprint3.MenuItem>
      </AcknowledgeWithCommentMenuItem>
    </ul>
  </Blueprint3.Menu>
</DisabledStationSohContextMenu>
`;

exports[`Station cell context menu StationSohContextMenu can be created 1`] = `
<StationSohContextMenu
  acknowledgeCallback={[MockFunction]}
  stationNames={
    Array [
      "test",
      "test2",
    ]
  }
>
  <Blueprint3.Menu>
    <ul
      className="bp3-menu"
    >
      <AcknowledgeMenuItem
        acknowledgeCallback={[MockFunction]}
        disabled={false}
        stationNames={
          Array [
            "test",
            "test2",
          ]
        }
      >
        <Blueprint3.MenuItem
          className="acknowledge-soh-status"
          data-cy="acknowledge-without-comment"
          disabled={false}
          multiline={false}
          onClick={[Function]}
          popoverProps={Object {}}
          shouldDismissPopover={true}
          text="Acknowledge 2 stations"
        >
          <li
            className=""
          >
            <a
              className="bp3-menu-item bp3-popover-dismiss acknowledge-soh-status"
              data-cy="acknowledge-without-comment"
              onClick={[Function]}
            >
              <Blueprint3.Icon />
              <Blueprint3.Text
                className="bp3-fill"
                ellipsize={true}
              >
                <div
                  className="bp3-text-overflow-ellipsis bp3-fill"
                >
                  Acknowledge 2 stations
                </div>
              </Blueprint3.Text>
            </a>
          </li>
        </Blueprint3.MenuItem>
      </AcknowledgeMenuItem>
      <AcknowledgeWithCommentMenuItem
        acknowledgeCallback={[MockFunction]}
        disabled={false}
        stationNames={
          Array [
            "test",
            "test2",
          ]
        }
      >
        <Blueprint3.MenuItem
          className="acknowledge-soh-status"
          data-cy="acknowledge-with-comment"
          disabled={false}
          multiline={false}
          onClick={[Function]}
          popoverProps={Object {}}
          shouldDismissPopover={true}
          text="Acknowledge 2 stations with comment..."
        >
          <li
            className=""
          >
            <a
              className="bp3-menu-item bp3-popover-dismiss acknowledge-soh-status"
              data-cy="acknowledge-with-comment"
              onClick={[Function]}
            >
              <Blueprint3.Icon />
              <Blueprint3.Text
                className="bp3-fill"
                ellipsize={true}
              >
                <div
                  className="bp3-text-overflow-ellipsis bp3-fill"
                >
                  Acknowledge 2 stations with comment...
                </div>
              </Blueprint3.Text>
            </a>
          </li>
        </Blueprint3.MenuItem>
      </AcknowledgeWithCommentMenuItem>
    </ul>
  </Blueprint3.Menu>
</StationSohContextMenu>
`;
